Apps/WebClient/src/ClientApp/app/components/timeline/commentSection.vue

```diff
@@ -174,5 +202,53 @@ export default class CommentSectionComponent extends Vue {
         this.isLoadingComments = false;
       });
   }
+
+  private updateComment(): void {
+    this.isLoadingComments = true;
+    let commentPromise = this.commentService
+      .updateComment({
+        id: this.editing.id,
+        text: this.commentInput,
+        userProfileId: this.editing.userProfileId,
+        parentEntryId: this.editing.parentEntryId,
+        createdDateTime: this.editing.createdDateTime,
+        version: this.editing.version
+      })
+      .then(result => {
+        this.getComments();
+        this.commentInput = "";
+      })
+      .catch(err => {
+        console.log(err);
+        this.hasErrors = true;
+      })
+      .finally(() => {
+        this.isLoadingComments = false;
+        this.showInput = false;
+        this.isEditMode = false;
+      });
+  }
+
+  private deleteComment(comment: UserComment): void {
+    if (confirm("Are you sure you want to delete this comment?")) {
```

Comment:
What are your thoughts on moving this to the comment component?

Reply:
Could do, but the parent would have to know when a comment has been deleted so it can retrieve a new list of comments and clear the text input. Also prefer to not pass the service to every comment component for simplicity's sake. Either way I think we'd be emitting from the child to the parent so I just kept all the service calls in the same component. Open to opinions though!